addons/binding/org.openhab.binding.mihome/src/main/java/org/openhab/binding/mihome/handler/MiHomeGatewayHandler.java

```diff
@@ -0,0 +1,575 @@
+            return subdevices;
+        }
+        return null;
+    }
+
+    /**
+     * Initializes the pairing process between subdevice and a gateway. Depending on the device the pairing could be
+     * initialized before or after pushing the pairing button on the device.
+     *
+     * @param gatewayID - the ID of the gateway
+     * @param deviceType - the device type, see {@link DeviceTypesConstants}
+     * @return - true if the paring was initialized or false if the request isn't successful
+     */
+    public boolean initializePairing(int gatewayID, String deviceType) {
+        JsonObject response = apiManager.registerSubdevice(gatewayID, deviceType);
+        if (response != null) {
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Sends a request to the Mi|Home server to unregister a subdevice. A success unfortunately doesn't guarantee that
+     * the device will be unregistered. If the connection to the gateway is lost the request will be successful, but it
+     * won't be completed when the connection is resumed
+     *
+     * @param subdeviceID - the ID of the subdevice to unregister
+     * @return - true if the request is sent or false if it has failed
+     */
+    public boolean unregisterSubdevice(int subdeviceID) {
+        JsonObject response = apiManager.unregisterSubdevice(subdeviceID);
+        if (response != null) {
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Gets a data for a subdevice
+     *
+     * @param subdeviceID - the ID of the subdevice
+     * @return JsonObject containing the requested data or null if the request isn't successful
+     */
+    public JsonObject getSubdeviceData(int subdeviceID) {
+        JsonObject response = apiManager.showSubdeviceInfo(subdeviceID);
+        if (response != null) {
+            JsonObject subdeviceInfo = (JsonObject) response.get(JSONResponseConstants.DATA_KEY);
+            return subdeviceInfo;
+        }
+        return null;
+    }
+
+    /**
+     * Updates a label of the subdevice in the MiHome API
+     *
+     * @param subdeviceID - the ID of the subdevice to update
+     * @param newLabel - the new label
+     * @return null if the request was successful, otherwise device data
+     */
+    public JsonObject updateSubdevice(int subdeviceID, String newLabel) {
+        JsonObject response = apiManager.updateSubdevice(subdeviceID, newLabel);
+        if (response != null) {
+            JsonObject subdeviceInfo = (JsonObject) response.get(JSONResponseConstants.DATA_KEY);
+            return subdeviceInfo;
+        }
+        return null;
+    }
+
+    public static String getFormattedCurrentDayTime() {
+        LocalDateTime curentLocalDateTime = LocalDateTime.now();
+        Date currentDate = Date.from(curentLocalDateTime.atZone(ZoneId.systemDefault()).toInstant());
+        return new SimpleDateFormat(MiHomeBindingConstants.LAST_SEEN_PROPERTY_PATTERN).format(currentDate);
```

Comment:
It does not seem to format a last_seen date but just today, the method is called `getFormattedCurrentDayTime`. The relation with last_seen is not clear (here).